pkg/prometheus/agent/operator.go

```diff
@@ -683,6 +713,29 @@ func (c *Operator) syncDaemonSet(ctx context.Context, key string, p *monitoringv
 		return nil
 	}
 
+	err = k8sutil.UpdateDaemonSet(ctx, dsetClient, dset)
+	sErr, ok := err.(*apierrors.StatusError)
+
```

Comment:
I've noticed that we don't compare input hashes for daemonsets while we do for statefulsets. The tests pass, so I'm not too worried, but I'm curious how this impacts the update frequency.

If this proves to be a real problem, it's ok to fix in a follow up PR

Reply:
Yeah we discussed some time ago in Slack, I and Simon both agreed to do this later (after MVP): https://cloud-native.slack.com/archives/C071STDT4FP/p1717580605023279